lib/connection.js

```diff
@@ -697,6 +697,18 @@ Connection.prototype.openUri = function(uri, options, callback) {
       typeof callback + '"');
   }
 
+  if (this._destroyCalled && this.readyState === STATES.disconnected) {
```

Comment:
I mean here you check if _destroyCalled is set and ff the readyState is equal to disconnected. Would it not make sense to set the state to STATES.destroyed if we called with destroy?

Reply:
Hmm, true @Uzlopak, till then the connection has been destroyed. This can be done I think. Currently, I will keep it as is, but @vkarpov15 could you provide information on if we can do this?

Reply:
@vkarpov15 Done with suggested changes